pgrx-macros/src/rewriter.rs

```diff
@@ -135,19 +136,44 @@ fn foreign_item_fn(func: &ForeignItemFn, abi: &syn::Abi) -> syn::Result<proc_mac
     let arg_list = rename_arg_list(&func.sig)?;
     let arg_list_with_types = rename_arg_list_with_types(&func.sig)?;
     let return_type = func.sig.output.clone();
+    let link_with_cshim = func.attrs.iter().any(|attr| match &attr.meta {
+        syn::Meta::NameValue(kv) if kv.path.get_ident().filter(|x| *x == "link_name").is_some() => {
+            if let Expr::Lit(ExprLit { lit: Lit::Str(value), .. }) = &kv.value {
+                value.value().ends_with("__pgrx_cshim")
+            } else {
+                false
+            }
+        }
+        _ => false,
+    });
+    let link = if link_with_cshim {
+        quote! {}
+    } else {
+        quote! { #[cfg_attr(target_os = "windows", link(name = "postgres"))] }
+    };
```

Comment:
Hrm. What should I be seeing and when? I am trying to force this to trigger while building the bindings and it doesn't seem to be happening?

Reply:
Ah, wait, this will only happen during macro expansion, sec...

Reply:
...which happens after `cfg`, hmmm...

Reply:
Okay, forced this through `cargo expand` after some fiddling and it works as expected.